presto-main/src/main/java/com/facebook/presto/metadata/StaticCatalogStore.java

```diff
@@ -75,25 +82,46 @@ public void loadCatalogs()
             }
         }
 
+        additionalCatalogs.forEach(this::loadCatalog);
+
         catalogsLoaded.set(true);
     }
 
     private void loadCatalog(File file)
             throws Exception
     {
         String catalogName = Files.getNameWithoutExtension(file.getName());
+
+        log.info("-- Loading catalog properties %s --", file);
+        Map<String, String> properties = new HashMap<>(loadProperties(file));
+        checkState(properties.containsKey("connector.name"), "Catalog configuration %s does not contain connector.name", file.getAbsoluteFile());
+
+        loadCatalog(catalogName, properties);
+    }
+
+    private void loadCatalog(String catalogName, Map<String, String> properties)
+    {
         if (disabledCatalogs.contains(catalogName)) {
             log.info("Skipping disabled catalog %s", catalogName);
             return;
         }
 
-        log.info("-- Loading catalog %s --", file);
-        Map<String, String> properties = new HashMap<>(loadProperties(file));
+        log.info("-- Loading catalog %s --", catalogName);
+
+        String connectorName = null;
+        ImmutableMap.Builder<String, String> connectorProperties = ImmutableMap.builder();
+        for (Entry<String, String> entry : properties.entrySet()) {
+            if (entry.getKey().equals("connector.name")) {
```

Comment:
why cannot we do 
```java
        String connectorName = properties.remove("connector.name");
```

Similar to the logic in https://github.com/prestodb/presto/blob/20aa3228be1317c74659be6a57003b79e738ae55/presto-main/src/main/java/com/facebook/presto/metadata/StaticCatalogStore.java#L93

Reply:
Now 

```
private void loadCatalog(String catalogName, Map<String, String> properties)
```

is used by the 

```
public void loadCatalogs(Map<String, Map<String, String>> additionalCatalogs)
```

where the `additionalCatalogs` parameter comes from the outside, and might be immutable.

Before a copy of the properties map was created here: https://github.com/prestodb/presto/pull/14152/files/4d8e7e658b13ce71c89f2fab515f5f461e640bb8#diff-665defb947f4d22b0cf16cb75130fcddR96

And actually, we don't have to create a copy any longer